app/services/current-session.js

```diff
@@ -38,21 +44,34 @@ export default class CurrentSessionService extends Service {
     this.user = null;
     this.role = null;
     this.organization = null;
+    this.impersonation.clear();
   }
 
   may(permission) {
     return this.userGroup?.permissions.includes(permission);
   }
 
+  reallyMay(permission) {
+    return this.ownUserGroup?.permissions.includes(permission);
+  }
+
   hasAccessToApplication() {
     return isPresent(this.role);
   }
 
   get userGroup() {
+    return this.impersonatedRole && findGroupByRole(this.impersonatedRole.uri);
```

Comment:
Shouldn't we fallback to `this.role` here in case the user is currently not impersonated?  Or am I missing something?